File: AppointmentService.py
from Appointment import Appointment

# Declaring variables, removing magic numbers
STORESAPPOINTMENT = 1

class AppointmentService:

    # Citation for self usage: (Gyanendra371, 2024)

    # (Erakshaya485, 2024), initializes the class.
    def __init__(self, addAppointment, deleteAppointment, appointmentList, appointmentId, appointmentDate, appointmentDescription):
        if addAppointment == None:
            return
        
        self.AppointmentService(addAppointment, deleteAppointment, appointmentList, appointmentId, appointmentDate, appointmentDescription)

    # Main appointment service function
    def AppointmentService(self, addAppointment, deleteAppointment, appointmentList, appointmentId, appointmentDate, appointmentDescription):

        self.added = False
        self.deleted = False
        appointmentExists = False

        if len(appointmentList) >= STORESAPPOINTMENT:

            # Checks if an appointment already exists with input id.
            for incrementAppointmentList in appointmentList:
                if appointmentId == incrementAppointmentList.getAppointmentId():
                    appointmentExists = True
        
                else:
                    appointmentExists = False
        
        # Adding appointment
        if addAppointment == True:
            confirmationAppointment = self.addingAppointment(addAppointment, appointmentExists, appointmentId, appointmentDate, appointmentDescription)
            
            # Test Confirmation
            self.added = True

            return confirmationAppointment

        # Deleting appointment
        if deleteAppointment == True:
            self.deletingAppointment(deleteAppointment, appointmentExists, appointmentList, appointmentId)
                


    # Adds an appointment
    def addingAppointment(self, addAppointment, appointmentExists, appointmentId, appointmentDate, appointmentDescription): 
        
        # (Kalra, 2023; Everythingtech, 2023), initializing new appointment
        newAppointment = Appointment(None, None, None, True)

        # (TonyA, 2013; Rollbar Editorial Team, 2023), adding appointment with error checking
        try:
            if (addAppointment == True) and (appointmentExists == False):
                newAppointment.Appointment(appointmentId, appointmentDate, appointmentDescription)
                return newAppointment
        
        # User incorrect id error check
            elif (addAppointment == True) and (appointmentExists == True):
                raise Exception
        except:
            print("Appointment with input id has already been made, please use another id.")

        
    # Deletes an appointment.
    def deletingAppointment(self, deleteAppointment, appointmentExists, appointmentList, appointmentId): 
        
        # Variable to iterate through appointment list to find location of same appointment
        sameAppointmentId = 0

        # (TonyA, 2013; Rollbar Editorial Team, 2023), deleting appointment with error checking
        try:
            if (deleteAppointment == True) and (appointmentExists == True):
                for incrementAppointmentList in appointmentList:
                    if appointmentId == incrementAppointmentList.appointmentId:
                        appointmentDelete = appointmentList.pop(sameAppointmentId)
                        appointmentDelete.deleteAppointmentInfo()
                        appointmentDelete = None
                        self.deleted = True
                
                    # Adds one after if statment since the list can store at [0]
                    sameAppointmentId += 1

            # User inccorect id error check
            elif (deleteAppointment == True) and (appointmentExists == False):
                raise Exception
        except:
                print("Appointment with input id does not exist, can not delete.")

        

File: Software Design and Engineering/Project One CS-320 Artifact - Ranen Hicks/Apppointment Service - Ranen Hicks/AppointmentService.java

import java.util.Date;

public class AppointmentService {
	
	public boolean added;
	public boolean deleted;
	
	// Constructor for appointment service object.
	public AppointmentService(boolean add, boolean delete, String id, Date date, String description) {
		
		// Makes an appointment for testing.
		Appointment oldAppointment = new Appointment("1", new Date(), "111");
		Appointment newAppointment = null;
		
		// Checks if a new appointment should be added.
		if (add == true && id != oldAppointment.getAppointmentId()) {
			newAppointment = new Appointment(id, date, description);
			added = true;
		}
		
		// The id was the same as a previous one so throws an error.
		else if (add == true) {
			throw new IllegalArgumentException("Id Already Has Appointment.");
		}
		
		// Checks if a new appointment should be deleted.
		if (delete == true) {
			
			// Deletes either of the appointments.
			if (added == true && id == newAppointment.getAppointmentId()) {
				newAppointment.deleteAppointment();
				newAppointment = null;
				deleted = true;
			}
			else if (id == oldAppointment.getAppointmentId()) {
				oldAppointment.deleteAppointment();
				oldAppointment = null;
				deleted = true;
			}
			
			// Throws error when there is no appointment to delete by that id
			else if (delete = true) {
				throw new IllegalArgumentException("No Appointment With Same Id");
			}
		}
	}
}
